tests/tests_lite/plugins/collectives/test_torch_collective.py

```diff
@@ -0,0 +1,237 @@
+    assert TorchCollective._convert_to_native_op("max") == ReduceOp.MAX
+    assert TorchCollective._convert_to_native_op("min") == ReduceOp.MIN
+    assert TorchCollective._convert_to_native_op("product") == ReduceOp.PRODUCT
+    assert TorchCollective._convert_to_native_op("sum") == ReduceOp.SUM
+    # Test we are passing through native ops without change
+    assert TorchCollective._convert_to_native_op(ReduceOp.BAND) == ReduceOp.BAND
+    assert TorchCollective._convert_to_native_op(ReduceOp.BOR) == ReduceOp.BOR
+    assert TorchCollective._convert_to_native_op(ReduceOp.BXOR) == ReduceOp.BXOR
+    assert TorchCollective._convert_to_native_op(ReduceOp.MAX) == ReduceOp.MAX
+    assert TorchCollective._convert_to_native_op(ReduceOp.MIN) == ReduceOp.MIN
+    assert TorchCollective._convert_to_native_op(ReduceOp.PRODUCT) == ReduceOp.PRODUCT
+    assert TorchCollective._convert_to_native_op(ReduceOp.SUM) == ReduceOp.SUM
+    # Test we are handling different casing properly
+    assert TorchCollective._convert_to_native_op("BOR") == ReduceOp.BOR
+    assert TorchCollective._convert_to_native_op("BoR") == ReduceOp.BOR
+
+    # AVG is very recent!
+    if _TORCH_GREATER_EQUAL_1_11:
+        assert TorchCollective._convert_to_native_op("avg") == ReduceOp.AVG
+
+    # Test invalid type
+    with pytest.raises(ValueError, match="op 1 should be a `str` or `ReduceOp`"):
+        TorchCollective._convert_to_native_op(1)
+
+    # Test invalid string
+    with pytest.raises(ValueError, match="op 'INVALID' is not a member of `ReduceOp`"):
+        TorchCollective._convert_to_native_op("invalid")
+
+
+@skip_distributed_unavailable
+def test_repeated_create_and_destroy():
+    collective = TorchCollective()
+    with mock.patch("torch.distributed.init_process_group"):
+        collective.setup()
+    with mock.patch("torch.distributed.new_group") as new_mock:
+        collective = TorchCollective(instantiate_group=True)
+    new_mock.assert_called_once()
+
+    with pytest.raises(RuntimeError, match="TorchCollective` already owns a group"):
+        collective.create_group()
+
+    with mock.patch("torch.distributed.destroy_process_group") as destroy_mock:
+        collective.teardown()
+    with pytest.raises(RuntimeError, match="TorchCollective` does not own a group to destroy"):
+        collective.teardown()
+
+    destroy_mock.assert_called_once_with(new_mock.return_value)
+    assert collective._group is None
+
+    with mock.patch("torch.distributed.new_group"), mock.patch("torch.distributed.destroy_process_group"):
+        # check we can create_group again
+        collective.create_group().teardown()
+
+
+def collective_launch(fn, parallel_devices, num_groups=1):
+    strategy = DDPSpawnStrategy(
+        accelerator=CPUAccelerator(), parallel_devices=parallel_devices, cluster_environment=LightningEnvironment()
+    )
+    launcher = _MultiProcessingLauncher(strategy=strategy)
+    collectives = [TorchCollective() for _ in range(num_groups)]
```

Comment:
TorchCollective() without specifying ranks=..., should it default to all ranks? We probably can't given the design, but would be worth checking what torch's create_group() does

Reply:
I think, it will default to all ranks on `TorchCollective` by default at this point and for any other collectives we decide to add, we can try to mimic this behavior as much as possible.